tests/test_api_dataset_folder.py

```diff
@@ -288,3 +299,30 @@ async def test_create_sub_folder_when_parent_folder_not_found_should_return_404(
     res = await client.post(f'/v1/dataset/{dataset_id}/folder', json=payload)
     assert res.status_code == 404
     assert res.json()['error'] == {'code': 'global.not_found', 'details': 'Requested resource is not found'}
+
+
+async def test_create_folder_with_long_folder_name_should_return_422(client, dataset, faker):
+    from random import choice
+    from string import ascii_lowercase
+
+    n = 31
+
+    folder_name = ''.join(choice(ascii_lowercase) for i in range(n))
```

Comment:
I suggest to use faker here to generate folder name, so we won't need to use above import.
```suggestion
    folder_name = faker.pystr(31, 31)
```